reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/UtilitiesTest.kt

```diff
@@ -0,0 +1,404 @@
+class UtilitiesTest {
+
+    private lateinit var stubSdkCore: StubSDKCore
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        stubSdkCore = StubSDKCore(forge)
+
+        val fakeTraceConfiguration = TraceConfiguration.Builder().build()
+        Trace.enable(fakeTraceConfiguration, stubSdkCore)
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        GlobalTracer::class.java.setStaticValue("isRegistered", false)
+    }
+
+    @RepeatedTest(16)
+    fun `M send span with exception W buildSpan() + start() + setError(Throwable) + finish()`(
+        @StringForgery fakeOperation: String,
+        @Forgery fakeThrowable: Throwable
+    ) {
+        // Given
+        stubSdkCore.stubFeature(Feature.LOGS_FEATURE_NAME)
+        val testedTracer = AndroidTracer.Builder(stubSdkCore).build()
+
+        // When
+        var traceId: String
+        var spanId: String
+        val fullDuration = measureNanoTime {
+            val span = testedTracer.buildSpan(fakeOperation).start()
+            traceId = span.traceIdAsHexString()
+            spanId = span.spanIdAsHexString()
+            Thread.sleep(OP_DURATION_MS)
+            span.setError(fakeThrowable)
+            span.finish()
+        }
+
+        // Then
+        val eventsWritten = stubSdkCore.eventsWritten(Feature.TRACING_FEATURE_NAME)
+        assertThat(eventsWritten).hasSize(1)
+        val event0 = JsonParser.parseString(eventsWritten[0].eventData) as JsonObject
+        println(event0)
+        assertThat(event0.getString("env")).isEqualTo(stubSdkCore.getDatadogContext().env)
+        assertThat(event0.getString("spans[0].trace_id")).isEqualTo(traceId)
+        assertThat(event0.getString("spans[0].span_id")).isEqualTo(spanId)
+        assertThat(event0.getString("spans[0].service")).isEqualTo(stubSdkCore.getDatadogContext().service)
+        assertThat(event0.getString("spans[0].meta.version")).isEqualTo(stubSdkCore.getDatadogContext().version)
+        assertThat(event0.getString("spans[0].meta._dd.source")).isEqualTo(stubSdkCore.getDatadogContext().source)
+        assertThat(event0.getString("spans[0].meta.tracer.version"))
+            .isEqualTo(stubSdkCore.getDatadogContext().sdkVersion)
+        assertThat(event0.getInt("spans[0].error")).isEqualTo(1)
+        assertThat(event0.getString("spans[0].name")).isEqualTo(fakeOperation)
+        assertThat(event0.getString("spans[0].resource")).isEqualTo(fakeOperation)
+        assertThat(event0.getLong("spans[0].duration")).isBetween(OP_DURATION_NS, fullDuration)
+        assertThat(event0.getString("spans[0].meta.error.type")).isEqualTo(fakeThrowable.javaClass.canonicalName)
+        assertThat(event0.getString("spans[0].meta.error.msg")).isEqualTo(fakeThrowable.message)
+        assertThat(event0.getString("spans[0].meta.error.stack")).isEqualTo(fakeThrowable.stackTraceToString())
+        val eventsReceived = stubSdkCore.eventsReceived(Feature.LOGS_FEATURE_NAME)
+        assertThat(eventsReceived).hasSize(0)
```

Comment:
minor: `isEmpty` will give more descriptive error

```suggestion
        assertThat(eventsReceived).isEmpty
```

Reply:
👍